### PR TITLE
Added `on_raw_typing` event

### LINKED ISSUE #7: [v2]: DM typing and Thread typing events are not dispatched.
### Link

https://github.com/Rapptz/discord.py/issues/7432

### PR DESCRIPTION
**Pull request by Astrea49**
## Summary

Adds in an `on_raw_typing` event. This may be useful if the typing starts in a DM, as otherwise, d.py is unable to resolve the user and so unable to dispatch an `on_typing` event.
Note that even with #145, this event is still useful in case the typing itself comes from a user that comes outside of the bot's user cache, a likely possibility if `Intents.members` is disabled. In fact, this PR was made and adjusted with #145 in mind, at least in terms of the documentation's wording (there might be merge conflicts between this PR and that one - sorry!).

Closes #7.

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue. (kind of? It's another solution to #7)
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Review by TAG-Epic — APPROVED** [review]

File: nextcord/state.py [hunk 1]
```diff
@@ -1330,28 +1330,37 @@ def parse_voice_server_update(self, data) -> None:
...
+            if user is not None:
+                self.dispatch('typing', channel, user, raw.when)
 
-                if member is None:
-                    member_data = data.get('member')
-                    if member_data:
-                        member = Member(data=member_data, state=self, guild=guild)
+    def _get_typing_user(self, channel: Optional[MessageableChannel], user_id: int) -> Optional[Union[User, Member]]:
+        if isinstance(channel, DMChannel):
+            return channel.recipient
```
Comment (veni-vidi-code) [inline nextcord/state.py:1355]:
```suggestion
            return channel.recipient or self.get_user(user_id)
```

**Review by veni-vidi-code — CHANGES_REQUESTED** [review]
This should do the trick of implementing both

**Review by ooliver1 — APPROVED** [review]
Assuming it's all tested, looks good to me

**Comment by Astrea49** [comment]
I've tested it with enhanced d.py and it works - there's no major difference between this version and edpy's version, so it should be fine.

**Comment by veni-vidi-code** [comment]
Maybe go ahead and mention the issue itself aswell to link it properly and close it when merging

**Review by veni-vidi-code — APPROVED** [review]

**Pull request merged**